core/src/main/java/com/valtech/aemsaas/core/services/configuration/BaseSaasConfiguration.java

```diff
@@ -0,0 +1,15 @@
+package com.valtech.aemsaas.core.services.configuration;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.caconfig.annotation.Configuration;
+import org.apache.sling.caconfig.annotation.Property;
+
+@Configuration(label = "Base AEM Saas - Configuration", description = "Base AEM Saas Configuration")
+public @interface BaseSaasConfiguration {
```

Comment:
general about naming: let's remove saas from our class names and replace it with search. BaseSaasConfiguration  --> BaseSearchConfiguration